respect-credentials/src/commonMain/kotlin/world/respect/credentials/passkey/request/CreatePublicKeyCredentialCreationOptionsJsonUseCase.kt

```diff
@@ -0,0 +1,74 @@
+package world.respect.credentials.passkey.request
+
+import world.respect.credentials.passkey.username.CreateCredentialUsernameUseCase
+import io.ktor.http.Url
+import io.ktor.util.encodeBase64
+import world.respect.credentials.passkey.model.AuthenticatorSelectionCriteria
+import world.respect.credentials.passkey.model.PublicKeyCredentialCreationOptionsJSON
+import world.respect.credentials.passkey.model.PublicKeyCredentialParameters
+import world.respect.credentials.passkey.model.PublicKeyCredentialRpEntity
+import world.respect.credentials.passkey.model.PublicKeyCredentialUserEntityJSON
+import world.respect.libutil.ext.randomString
+
+/**
+ * Create the Json that is used to request creation of a new passkey. This should work on all
+ * platforms where passkeys are supported (Android, Web, etc).
+ *
+ * As per https://w3c.github.io/webauthn/#dictdef-publickeycredentialcreationoptionsjson
+ *
+ * The passkey rpId will always be the SystemUrlConfig.systemBaseUrl host
+ * The passkey user.id will be a unique 64bit long.
+ *
+ * The userHandle (PublicKeyCredentialUserEntityJSON.id) is encoded to include the person passkey
+ * UID and Learning Space URL - see EncodeUserHandleUseCase
+ */
+class CreatePublicKeyCredentialCreationOptionsJsonUseCase(
+    private val url:Url,
```

Comment:
Change url to rpId - because the rpId will be known at the time of login / requesting credentials.

Reply:
Type should be changed to String.